compiler/src/dotty/tools/dotc/inlines/Inlines.scala

```diff
@@ -443,6 +443,9 @@ object Inlines:
             unrollTupleTypes(tail).map(head :: _)
           case tpe: TermRef if tpe.symbol == defn.EmptyTupleModule =>
             Some(Nil)
+          case tpRef: TypeRef => tpRef.info match
```

Comment:
The fix works, but I am not sure if this is the correct way to handle `MatchAlias`es.

@sjrd could you have a look at this one?

Reply:
It looks reasonable, although usually we would rather expect
```scala
case tpe: AppliedType if tpe.isMatchAlias => ... tpe.tryNormalize ...
```
It is a bit suspicious that we directly encounter the `tycon` here, rather than the surrounding `AppliedType`. What are `tpe` and `tpe.dealias` when we get here on the problematic test case?

Reply:
Yes I noticed that for instance `isMatchAlias` is defined on `AppliedType`s. 
But in this case, we have `tpe := TypeRef(NoPrefix, Alias)` which remains the same after `dealias` since the info of `Alias` is not a `TypeAlias` but a `MatchAlias`. The case is not covered here: 
https://github.com/lampepfl/dotty/blob/997103d818ead17d528d320b01edefb8792579ac/compiler/src/dotty/tools/dotc/core/Types.scala#L1438-L1445